### PR TITLE
8255742: PrintInlining as compiler directive doesn't print virtual calls

### PR DESCRIPTION
**Pull request by simonis**
If we set PrintInlining for a specific method only with the help of a compiler directive (e.g. `-XX:CompileCommand="option,Inlining::foo,PrintInlining"`) it won't print virtual calls which can't be inlined. These calls are printed if we use the global `-XX:+PrintInlining` option.

The fix is trivial:
```
--- a/src/hotspot/share/opto/doCall.cpp
+++ b/src/hotspot/share/opto/doCall.cpp
@@ -357,7 +357,7 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
   // Use a more generic tactic, like a simple call.
   if (call_does_dispatch) {
     const char* msg = "virtual call";
- if (PrintInlining) print_inlining(callee, jvms->depth() - 1, jvms->bci(), msg);
+ if (C->print_inlining()) print_inlining(callee, jvms->depth() - 1, jvms->bci(), msg);
     C->log_inline_failure(msg);
     return CallGenerator::for_virtual_call(callee, vtable_index);
   } else {
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255742](https://bugs.openjdk.java.net/browse/JDK-8255742): PrintInlining as compiler directive doesn't print virtual calls

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to c26db9a16d635a6d60f9a100ddfdc65f4e5765b7
 * @eastig (no known github.com user name / role) ⚠️ Review applies to c26db9a16d635a6d60f9a100ddfdc65f4e5765b7
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1596/head:pull/1596`
`$ git checkout pull/1596`

### CODE REVIEW

File: test/hotspot/jtreg/compiler/inlining/PrintInlining.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2015, 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/inlining/PrintInlining.java:2]: Not sure if the copyright should really mention 2015 :)

**Review by TobiHartmann — APPROVED** [review]
Otherwise looks good to me.

**Review by eastig — APPROVED** [review]
LGTM

File: test/hotspot/jtreg/compiler/inlining/PrintInlining.java (hunk 1 above)
Comment (simonis) [reply test/hotspot/jtreg/compiler/inlining/PrintInlining.java]: Good catch!

I copied the header and *specially* updated the second copyright year (but forgot that we only need one for a new file :))

**Comment by simonis** [comment]
> LGTM

Thanks Evgeny!

**Review by vnkozlov — APPROVED** [review]

**Comment by simonis** [comment]
Thanks Vladimir!

File: test/hotspot/jtreg/compiler/inlining/PrintInlining.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                .filter(s -> s.matches(".*A::bar.+virtual call.*"))
+                .count() != 1) {
+                throw new Exception("'" + option + "' didn't print virtual call.");
+            }
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        test("-XX:+PrintInlining");
+        test("-XX:CompileCommand=option,*::foo,PrintInlining");
```
Comment (navyxliu) [inline test/hotspot/jtreg/compiler/inlining/PrintInlining.java:67]: just FYI, even thought the old form is still supported,  the new form is -XX:CompileCommand=<option><method pattern>, ie. -XX:CompileCommand=PrintInlining,*::foo

Comment (simonis) [reply test/hotspot/jtreg/compiler/inlining/PrintInlining.java:67]: Thanks Xin! I indeed didn't knew that.

You make me feel so old :)

**Pull request closed without merging**